File: Algorithm-i/rotate.py
class Solution:
    def rotate(self, nums: list[int], k: int) -> None:
        last = len(nums)
        i = 0

        while i < k:
            for j in reversed(range(1, last)):
                temp = nums[j]
                nums[j] = nums[j - 1]
                nums[j - 1] = temp
            i += 1
        
        print(nums)

sol = Solution()

sol.rotate([7,9,6,5,3,0,8,7], 3)

File: Algorithm-i/rotate.cpp
#include <iostream>
#include <vector>

using std::cout;
using std::vector;

class Solution
{
public:
    void rotate(vector<int> &nums, int k)
    {
        int last_in = nums.size() - 1;
        int i = 0;

        while (i <= k)
        {
            for (int j = 0; j < nums.size(); j++)
            {
                nums[i] = nums[i - 1];
            }
            i++;
        }
        for (auto x : nums)
        {
            cout << "," << x;
        }
    }
};

int main()
{
    Solution sol;
    vector<int> nums = {1, 2, 3, 4, 5, 6, 7};
    sol.rotate(nums, 3);
    return 0;
}